driver/csiplugin/controllerserver.go

```diff
@@ -491,6 +491,29 @@ func (cs *ScaleControllerServer) getConnFromClusterID(cid string) (connectors.Sp
 	return nil, status.Error(codes.Internal, fmt.Sprintf("unable to find cluster [%v] details in custom resource", cid))
 }
 
+// checkSCSupportedParams checks if given CreateVolume request parameter keys
+// are supported by Spectrum Scale CSI and returns ("", true) if all parameter
+// keys are supported, otherwise returns (<invalid parameter key>, false)
+func checkSCSupportedParams(params map[string]string) (string, bool) {
+	isValidParam := false
+	for k := range params {
+		switch k {
+		case "csi.storage.k8s.io/pv/name", "csi.storage.k8s.io/pvc/name",
+			"csi.storage.k8s.io/pvc/namespace", "storage.kubernetes.io/csiProvisionerIdentity",
+			"volBackendFs", "volDirBasePath", "uid", "gid", "permissions",
+			"clusterId", "filesetType", "parentFileset", "inodeLimit",
+			"version", "tier", "compression":
+			isValidParam = true
+		default:
+			isValidParam = false
+		}
+		if !isValidParam {
+			return k, isValidParam
```

Comment:
Can we error out with all the invalid parameter in one go instead of failing one at a time

Reply:
Done